main/src/core/MonoDevelop.Core/MonoDevelop.Projects/SdkProjectExtension.cs

```diff
@@ -266,86 +267,108 @@ internal protected override async Task OnReevaluateProject (ProgressMonitor moni
 		}
 
 		/// <summary>
-		/// Returns all transitive references.
+		/// ASP.NET Core projects have different build actions if the file is in the wwwroot folder.
+		/// It also uses Content build actions for *.json, *.config and *.cshtml files. To support
+		/// this the default file globs for the file are found and the MSBuild item name is returned.
 		/// </summary>
-		internal protected override async Task<List<AssemblyReference>> OnGetReferences (
-			ConfigurationSelector configuration,
-			System.Threading.CancellationToken token)
+		internal protected override string OnGetDefaultBuildAction (string fileName)
 		{
-			var references = new List<AssemblyReference> ();
-
-			var traversedProjects = new HashSet<string> (StringComparer.OrdinalIgnoreCase);
-			traversedProjects.Add (Project.ItemId);
+			string include = MSBuildProjectService.ToMSBuildPath (Project.ItemDirectory, fileName);
+			var globItems = Project.MSBuildProject.FindGlobItemsIncludingFile (include).ToList ();
```

Comment:
Maybe we can do:
```suggestion
			var globItems = Project.MSBuildProject.FindGlobItemsIncludingFile (include).SingleOrDefault ();
```

To avoid matching all the items.

Reply:
Not sure I can change this. There may not be any matches - but could use FirstOrDefault instead. If there is more than one match we do not know what to return so fallback to asking the project for its default build action. Although not sure I have seen any examples of multiple matches here.

Reply:
Yeah, was just trying to avoid a new collection allocation, as it seems redundant.